src/commonTest/kotlin/app/cash/turbine/CustomRuntimeException.kt

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (C) 2020 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.turbine
+
+fun customRuntimeException(message: String? = null) : RuntimeException {
+  // Use a custom subtype to prevent coroutines from breaking referential equality.
+  return CustomRuntimeException(message)
+}
```

Comment:
Might as well inline this fn?

Reply:
I used it to keep the type private so it wouldn't be reflectively constructed. We don't have to do that for the assertion error type, though. Maybe I just need to mess with the signature a bit more.

Reply:
Yeah I had to adjust the signature a bit so coroutines wouldn't break references, but it's `internal` now so it can just be referenced directly.